### PR TITLE
Ensure only plugin REST tests are run for plugins

### LINKED ISSUE #53183: Plugin development - extra REST tests are executed
Starting with 7.6.1 some [internal changes](https://github.com/elastic/elasticsearch/pull/52114) to how the REST tests are managed accidentally causes all of the REST tests (not just the plugin tests) to execute. This applies to any plugin that uses the REST test framework from build-tools. The plugin REST tests are still run. 

The workaround is to add the following to `build.gradle` after applying the plugins:
```
restResources {
  restApi {
    includeCore '*'
  }
}
```

### PR DESCRIPTION
**Pull request by jakelandis**
This commit fixes ensures that for external builds
(e.g. plugin development) that the REST tests that are
copied are properly filtered to only include the API
by default.

The code prior to this change resulted in including both
the API and tests since the copy.include resulted as an
empty list by default since the stream is empty unless
explictly configured.

related #52114
fixes #53183

----------

Note - this is how I tested

```
# from elasticsearch directory
./gradlew :build-tools:publishToMavenLocal

./gradlew :rest-api-spec:publishToMavenLocal :test:framework:publishToMavenLocal :server:publishToMavenLocal :libs:elasticsearch-core:publishToMavenLocal :libs:elasticsearch-x-content:publishToMavenLocal :libs:elasticsearch-cli:publishToMavenLocal :libs:elasticsearch-nio:publishToMavenLocal :libs:elasticsearch-geo:publishToMavenLocal :libs:elasticsearch-secure-sm:publishToMavenLocal :distribution:archives:integ-test-zip:publishToMavenLocal :client:rest:publishToMavenLocal :client:sniffer:publishToMavenLocal :test:logger-usage:publishToMavenLocal


```

We can use [@spinscale 's cookie cutter](https://github.com/spinscale/cookiecutter-elasticsearch-ingest-processor) to help create a test project
```
# requires python3
pip3 install cookiecutter
cookiecutter gh:spinscale/cookiecutter-elasticsearch-ingest-processor
# Add the master version when it asks: elasticsearch_version [7.6.1]: 8.0.0-SNAPSHOT
cd ingest-awesome
# Add to the build.gradle
repositories {
   mavenLocal()
}
./gradlew check 
```

### CODE REVIEW

File: buildSrc/src/main/java/org/elasticsearch/gradle/test/rest/CopyRestApiTask.java [hunk 1]
```diff
@@ -132,7 +132,11 @@ void copy() {
             project.copy(c -> {
                 c.from(project.zipTree(coreConfig.getSingleFile()));
                 c.into(getTestSourceSet().getOutput().getResourcesDir()); // this ends up as the same dir as outputDir
-                c.include(includeCore.get().stream().map(prefix -> COPY_TO + "/" + prefix + "*/**").collect(Collectors.toList()));
+                if (includeCore.get().isEmpty()) {
+                    c.include(COPY_TO + "/**");
```
Comment (mark-vieira) [inline buildSrc/src/main/java/org/elasticsearch/gradle/test/rest/CopyRestApiTask.java]: The fact we are calling the path prefix for api specs `COPY_TO` I think makes this hard to follow. I'm thinking "why are we referencing the copy _destination_ when filtering _sources_?". If we are going to use this constant for both purposes, we should rename it appropriately, something like `API_SPEC_PREFIX`.

**Review by mark-vieira — APPROVED** [review]
One minor comment on naming, otherwise LGTM.

File: buildSrc/src/main/java/org/elasticsearch/gradle/test/rest/CopyRestApiTask.java (hunk 1 above)
Comment (jakelandis) [reply buildSrc/src/main/java/org/elasticsearch/gradle/test/rest/CopyRestApiTask.java]: Renamed `COPY_TO` to `REST_API_PREFIX` and `REST_TEST_PREFIX` on  https://github.com/elastic/elasticsearch/pull/53184/commits/f8459c9420d05ec90b70bbc75401ea5e1d84254e
